### PR TITLE
Fix #3504: Improve rendering of money effect

### LINKED ISSUE #3504: Floating money bug (graphic bug)
**OS:** windows 10
**Version:** 0.0.5
**Commit:** a937a65b286e264e15e0b793aae8b998ce25d935

scr11

scr13

**Steps to reproduce**
1. Build anything that has cost (for example: footpaths).
2. See the bug.

Apparently this happens since the PR #3444 was merged into develop.

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

File: src/paint/paint.c [hunk 1]
```diff
@@ -429,9 +429,9 @@ void sub_685EBC(money32 amount, uint16 string_id, sint16 y, sint16 z, sint8 y_of
 	ps->string_id = string_id;
 	ps->next = 0;
 	ps->args[0] = amount;
-	ps->args[4] = y;
-	ps->args[8] = y_offsets;
-	ps->args[12] = 0;
+	ps->args[1] = y;
+	ps->args[2] = y_offsets;
+	ps->args[3] = NULL;
```
Comment (LinusU) [inline src/paint/paint.c]: Since `args[3]` is a `uint32`, doesn't `0` make more sense than `NULL`?

Comment (marijnvdwerf) [inline src/paint/paint.c]: I have absolutely no idea.

File: src/paint/paint.c [hunk 2]
```diff
@@ -429,9 +429,9 @@ void sub_685EBC(money32 amount, uint16 string_id, sint16 y, sint16 z, sint8 y_of
 	ps->string_id = string_id;
 	ps->next = 0;
 	ps->args[0] = amount;
-	ps->args[4] = y;
-	ps->args[8] = y_offsets;
-	ps->args[12] = 0;
+	ps->args[1] = y;
+	ps->args[2] = y_offsets;
```
Comment (duncanspumpkin) [inline src/paint/paint.c]: Can you add casts to supress the warnings. Ps. This is going to mess up on x64 I expect.

Comment (marijnvdwerf) [inline src/paint/paint.c]: @janisozaur is free to take over.

**Pull request merged**